### PR TITLE
Add support for tokens

### PR DESCRIPTION
**Pull request by awrichar**

### CODE REVIEW

**Comment by awrichar** [comment]
Added a possible token contract based on ERC-1155. See also #129 for background on contract exploration.

File: solidity_firefly/contracts/ERC1155MixedFungible.sol [hunk 1]
```diff
@@ -0,0 +1,92 @@
+pragma solidity ^0.6.0;
```
Comment (peterbroadhurst) [inline solidity_firefly/contracts/ERC1155MixedFungible.sol:1]: Could you pop a license header in this one?

**Review by peterbroadhurst — COMMENTED** [review]
Great to see the progress here - with the asset manager periodic table element coming to life.

File: db/migrations/postgres/000024_create_token_tables.up.sql [hunk 2]
```diff
@@ -0,0 +1,12 @@
+BEGIN;
+CREATE TABLE tokenpool (
+  seq            SERIAL          PRIMARY KEY,
+  pool_id        VARCHAR(80)     NOT NULL,
```
Comment (awrichar) [inline db/migrations/postgres/000024_create_token_tables.up.sql:4]: pool_id should be a FireFly-generated UUID
on-chain ID should be another field (protocol_id?) - see transactions table for one example

File: db/migrations/postgres/000024_create_token_tables.up.sql [hunk 3]
```diff
@@ -7,6 +7,13 @@ CREATE TABLE tokenpool (
 );
 
 CREATE UNIQUE INDEX tokenpool_id ON tokenpool(pool_id);
-CREATE UNIQUE INDEX tokenpool_uri ON tokenpool(base_uri);
+
+CREATE TABLE tokenaccount (
+  seq            SERIAL,
+  member         VARCHAR(1024)   NOT NULL,
+  pool_id        VARCHAR(80)     NOT NULL,
```
Comment (awrichar) [inline db/migrations/postgres/000024_create_token_tables.up.sql:14]: Use the FireFly UUID for pool

File: db/migrations/postgres/000024_create_token_tables.up.sql [hunk 4]
```diff
@@ -7,6 +7,13 @@ CREATE TABLE tokenpool (
 );
 
 CREATE UNIQUE INDEX tokenpool_id ON tokenpool(pool_id);
-CREATE UNIQUE INDEX tokenpool_uri ON tokenpool(base_uri);
+
+CREATE TABLE tokenaccount (
+  seq            SERIAL,
+  member         VARCHAR(1024)   NOT NULL,
```
Comment (awrichar) [inline db/migrations/postgres/000024_create_token_tables.up.sql:13]: Should be the on-chain address (check naming)

Comment (awrichar) [reply db/migrations/postgres/000024_create_token_tables.up.sql:13]: `onchain_id`

File: db/migrations/postgres/000024_create_token_tables.up.sql (hunk 3 above)
Comment (awrichar) [reply db/migrations/postgres/000024_create_token_tables.up.sql:14]: Also need another field for the token_id (relevant for non-fungible)

**Comment by awrichar** [comment]
After discussing with some stakeholders, we've decided to split some of this functionality out as a separate micro-service here: https://github.com/hyperledger-labs/firefly-tokens-erc1155
Everything specific to ERC-1155 and the formulation of token IDs will be placed in that repository for better modularity, with a more generic REST API between it and FireFly for creating Token Pools and Accounts.

**Comment by awrichar** [comment]
The initial PR for the tokens shim service is posted: https://github.com/hyperledger-labs/firefly-tokens-erc1155/pull/2

Closing this PR and will open a new one to plug into the micro-service once it has shaped up further.

**Comment by awrichar** [comment]
This is now replaced by #154

**Pull request closed without merging**